.devcontainer/devcontainer.json

```diff
@@ -2,42 +2,88 @@
 {
-	"name": "Existing Docker Compose (Extend)",
+  "name": "homeru kun",
 
-	// Update the 'dockerComposeFile' list if you have more compose files or use different names.
-	// The .devcontainer/docker-compose.yml file contains any overrides you need/want to make.
-	"dockerComposeFile": [
-		"../docker-compose.yml",
-	],
+  // Update the 'dockerComposeFile' list if you have more compose files or use different names.
+  // The .devcontainer/docker-compose.yml file contains any overrides you need/want to make.
+  "dockerComposeFile": ["../docker-compose.yml"],
 
-	// The 'service' property is the name of the service for the container that VS Code should
-	// use. Update this value and .devcontainer/docker-compose.yml to the real service name.
-	"service": "homeru_bot",
+  // The 'service' property is the name of the service for the container that VS Code should
+  // use. Update this value and .devcontainer/docker-compose.yml to the real service name.
+  "service": "homeru_bot",
 
-	// The optional 'workspaceFolder' property is the path VS Code should open by default when
-	// connected. This is typically a file mount in .devcontainer/docker-compose.yml
-	"workspaceFolder": "/homeru_bot",
+  // The optional 'workspaceFolder' property is the path VS Code should open by default when
+  // connected. This is typically a file mount in .devcontainer/docker-compose.yml
+  "workspaceFolder": "/homeru_bot",
 
-	// Set *default* container specific settings.json values on container create.
-	"settings": {
-		"terminal.integrated.shell.linux": null
-	},
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  // "forwardPorts": [],
 
-	// Add the IDs of extensions you want installed when the container is created.
-	"extensions": []
+  // Uncomment the next line if you want start specific services in your Docker Compose config.
+  // "runServices": [],
 
-	// Use 'forwardPorts' to make a list of ports inside the container available locally.
-	// "forwardPorts": [],
+  // Uncomment the next line if you want to keep your containers running after VS Code shuts down.
+  "shutdownAction": "stopCompose",
 
-	// Uncomment the next line if you want start specific services in your Docker Compose config.
-	// "runServices": [],
+  // Uncomment the next line to run commands after the container is created - for example installing curl.
+  // "postCreateCommand": "apt-get update && apt-get install -y curl",
 
-	// Uncomment the next line if you want to keep your containers running after VS Code shuts down.
-	// "shutdownAction": "none",
+  // Uncomment to connect as a non-root user. See https://aka.ms/vscode-remote/containers/non-root.
+  "remoteUser": "vscode",
+  "postCreateCommand": "pip install -U pip && pip install -r requirements.txt -r requirements-dev.txt",
+  "postAttachCommand": "pip install -U pip && pip install -r requirements.txt -r requirements-dev.txt",
```

Comment:
これ`postCreateCommand`あるのになぜ定義されているのでしょうか？
公式には

> The parameters behave exactly like postCreateCommand, but the commands execute on attach rather than create. Not set by default.

とあるので、`postCreateCommand`と同じように実行されるが、発火タイミングが異なると認識しております！

Reply:
postAttachCommandは起動しているコンテナに割り当てられた場合に発火します。
以前、調べた時に必要だと思ったのですが、パッと出てきません。申し訳ありません。
調査時の資料を見てから再度コメントいたします。

Reply:
すいません。
以前調査時の資料を確認したところ不要でした。ので `postAttachCommand` は削除いたします。 🙇